packages/swap/src/utils/getPoolQuote.ts

```diff
@@ -41,14 +48,23 @@ export default async function getPoolQuote({
   // https://linear.app/chainflip/issue/PRO-1370/include-boost-fees-in-quote-from-cf-swap-rate-v2
   if (boostFeeBps) {
     const boostFee = getPipAmountFromAmount(depositAmount, boostFeeBps);
-    includedFees.push(buildFee(srcAsset, 'BOOST', boostFee));
+    includedFees.push({
+      type: 'BOOST',
+      amount: boostFee,
+      ...internalAssetToRpcAsset[srcAsset],
+    } as const);
     cfRateInputAmount -= boostFee;
   }
 
   if (isVaultSwap) {
     excludeFees.push('IngressDepositChannel');
   }
 
+  if (isOnChain) {
+    excludeFees.push('Egress');
+    excludeFees.push('IngressDepositChannel');
```

Comment:
do we need to default brokerCommission to 0 for `onChainSwaps` ?

Reply:
added a check to the schema validation